samplelistener/listener/dispatcher_service.bal

```diff
@@ -2,46 +2,43 @@ import ballerina/http;
 import ballerina/jballerina.java;
 
 service class DispatcherService {
-   private SlackAppMentionHandlingService|SlackAppCreatedHandlingService serviceRef;
-   //private string verificationToken;
+     private map<GenericService> services = {};
 
-   isolated function init(SlackAppMentionHandlingService|SlackAppCreatedHandlingService serviceRef) returns error? { //Receive verification token
-        self.serviceRef = serviceRef;
-        //self.verificationToken = "9asdbas9009123nas1e2";
+   isolated function addServiceRef(string serviceType, GenericService genericService) returns error? {
+        if (self.services.hasKey(serviceType)) {
+             return error("Same service type has been defined more than one time");
+        }
+        self.services[serviceType] = genericService;
    }
 
-   isolated resource function post events (http:Caller caller, http:Request request) returns error? {
+   // We are not using the (@http:payload GenericEventWrapperEvent g) notation because of 
+   // a bug in Ballerina. (Unable to handle complex objects as payloads)
```

Comment:
Rather than inability to handle complex objects it's an issue specific to service classes?